iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/BaseHiveIcebergMetaHook.java

```diff
@@ -0,0 +1,437 @@
+            "Iceberg table already created - can not use provided partition specification");
+
+        LOG.info("Iceberg table already exists {}", icebergTable);
+        return;
+      } catch (NoSuchTableException nte) {
+        // If the table does not exist we will create it below
+      }
+    }
+
+    // If the table does not exist collect data for table creation
+    // - InputFormatConfig.TABLE_SCHEMA, InputFormatConfig.PARTITION_SPEC takes precedence so the user can override the
+    // Iceberg schema and specification generated by the code
+
+    Set<String> identifierFields = Optional.ofNullable(request.getPrimaryKeys())
+        .map(primaryKeys ->
+            primaryKeys.stream().map(SQLPrimaryKey::getColumn_name).collect(Collectors.toSet()))
+        .orElse(Collections.emptySet());
+
+    Schema schema = schema(catalogProperties, hmsTable, identifierFields);
+    PartitionSpec spec = spec(conf, schema, hmsTable);
+
+    // If there are partition keys specified remove them from the HMS table and add them to the column list
+    if (hmsTable.isSetPartitionKeys()) {
+      hmsTable.getSd().getCols().addAll(hmsTable.getPartitionKeys());
+      hmsTable.setPartitionKeysIsSet(false);
+    }
+
+    catalogProperties.put(InputFormatConfig.TABLE_SCHEMA, SchemaParser.toJson(schema));
+    String specString = PartitionSpecParser.toJson(spec);
+    catalogProperties.put(InputFormatConfig.PARTITION_SPEC, specString);
+    validateCatalogConfigsDefined();
+
+    if (request.getEnvContext() == null) {
+      request.setEnvContext(new EnvironmentContext());
+    }
+    request.getEnvContext().putToProperties(TableProperties.DEFAULT_PARTITION_SPEC, specString);
+    setCommonHmsTablePropertiesForIceberg(hmsTable);
+
+    if (hmsTable.getParameters().containsKey(BaseMetastoreTableOperations.METADATA_LOCATION_PROP)) {
+      createHMSTableInHook = true;
+    }
+
+    assertFileFormat(catalogProperties.getProperty(TableProperties.DEFAULT_FILE_FORMAT));
+
+    // Set whether the format is ORC, to be used during vectorization.
+    setOrcOnlyFilesParam(hmsTable);
+    // Remove hive primary key columns from table request, as iceberg doesn't support hive primary key.
+    request.setPrimaryKeys(null);
+    setSortOrder(hmsTable, schema, catalogProperties);
+  }
+
+  /**
+   * Method for verification that necessary catalog configs are defined in Session Conf.
+   *
+   * <p>If the catalog name is provided in 'iceberg.catalog' table property,
+   * and the name is not the default catalog and not hadoop catalog, checks that one of the two configs
+   * is defined in Session Conf: iceberg.catalog.<code>catalogName</code>.type
+   * or iceberg.catalog.<code>catalogName</code>.catalog-impl. See description in Catalogs.java for more details.
+   *
+   */
+  private void validateCatalogConfigsDefined() {
+    Optional.ofNullable(catalogProperties.getProperty(InputFormatConfig.CATALOG_NAME))
+        .filter(name -> !StringUtils.isEmpty(name))
```

Comment:
i think this a bit overkill
````
String catalogName = catalogProperties.getProperty(InputFormatConfig.CATALOG_NAME);
if (!StringUtils.isEmpty(catalogName) && Catalogs.ICEBERG_HADOOP_TABLE_NAME.equals(catalogName)) {
  .....
}
````

Reply:
Reverted to non-functional style as before.